### PR TITLE
fix(blob): enforce maximumSizeInBytes client-side for multipart uploads

### LINKED ISSUE #873: maximumSizeInBytes not enforced for multipart uploads
Description:
When using multipart uploads, the maximumSizeInBytes setting is not respected. Files are uploaded successfully regardless of their size, even when they exceed the configured limit.

Steps to Reproduce:
1. Configure an upload with maximumSizeInBytes set (e.g. 10_000_000).
2. Upload a file larger than the configured maximum using multipart upload.
3. The upload completes successfully instead of being blocked.

Expected Behavior:
The upload should be rejected once the file size exceeds the maximumSizeInBytes limit.

Actual Behavior:
The upload completes successfully regardless of the file size when multipart uploads are used.

### PR DESCRIPTION
**Pull request by matingathani**
## Summary

Fixes #873 — `maximumSizeInBytes` was not enforced for multipart uploads.

`uncontrolledMultipartUpload` never checked `maximumSizeInBytes`. For bodies with a known size (Blob, File, Buffer), a client-side check is now done **before** the multipart upload begins, avoiding wasted API calls (create + upload parts) that would ultimately be rejected by the server.

Streams are skipped since their size is unknown upfront — enforcement is left to the server.

Also adds `maximumSizeInBytes` to `CommonCreateBlobOptions` so it's available to the multipart code path.

## Test plan

- New test: `multipart: true` + `maximumSizeInBytes` throws before upload starts
- New test: stream body with `maximumSizeInBytes` does not throw client-side

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by matingathani** [comment]
Addressed Copilot's review comments in the follow-up commit (ee4224d): changed the options type to `CommonCreateBlobOptions & WithUploadProgress` to resolve the type mismatch, tightened the JSDoc to accurately reflect multipart-only enforcement, and fixed the empty-string edge case in the body length guard. All 105 tests still passing locally.

File: packages/blob/src/helpers.ts [hunk 1]
```diff
@@ -64,6 +64,13 @@ export interface CommonCreateBlobOptions extends BlobCommandOptions {
    * If the ETag doesn't match, a `BlobPreconditionFailedError` will be thrown.
    */
   ifMatch?: string;
+  /**
+   * Maximum size in bytes allowed for this upload. Applies to both regular and
+   * multipart uploads. For bodies with a known size (Blob, File, Buffer, etc.)
+   * the check is performed client-side before the upload starts. The maximum
+   * allowed value is 5TB.
```
[the remark this answers is not in the document]
Comment (vvo) [reply packages/blob/src/helpers.ts]: Already addressed — the JSDoc was narrowed in the follow-up commit (ee4224d) and the type was moved to `CommonCreateBlobOptions` which is the correct scope for multipart.

File: packages/blob/src/multipart/uncontrolled.ts [hunk 2]
```diff
@@ -21,14 +21,30 @@ export async function uncontrolledMultipartUpload(
...
+
+  // For bodies with a known size (Blob, File, Buffer, etc.) enforce
+  // maximumSizeInBytes client-side before starting the upload. This avoids
+  // creating a multipart upload that will ultimately fail.
+  // Streams are skipped because their size is unknown upfront.
+  if (
+    options.maximumSizeInBytes !== undefined &&
+    totalToLoad > 0 &&
+    !isStream(body as Parameters<typeof isStream>[0]) &&
+    totalToLoad > options.maximumSizeInBytes
```
[the remark this answers is not in the document]
Comment (vvo) [reply packages/blob/src/multipart/uncontrolled.ts]: Already fixed — the function signature was updated to `CommonCreateBlobOptions & WithUploadProgress` in the follow-up commit (ee4224d).

**Comment by vvo** [comment]
@matingathani Thanks. We just updated our Blob API to allow for empty files to be uploaded. I kept your addition as for making max size check locally on multipart uploads.

**Comment by matingathani** [comment]
Thanks for merging and for the context! Really glad the multipart size check was useful. I'd love to keep contributing to this repo , I am  happy to pick up issues or dive into areas where help is needed. Looking forward to learning more about how the Blob API works under the hood.
